File: bench/elixir/primes.exs
defmodule B do
  def prime?(n) when n < 2, do: false
  def prime?(n), do: check(n, 2)
  defp check(n, d) when d * d > n, do: true
  defp check(n, d), do: if(rem(n, d) == 0, do: false, else: check(n, d + 1))
end

n = String.to_integer(System.get_env("BENCH_N") || "20000")
IO.puts(Enum.count(2..(n - 1), &B.prime?/1))

File: bench/erlang/primes.erl
#!/usr/bin/env escript
%%! -mode(compile)
main(_) ->
    N = bench_n(150000),
    Count = length([K || K <- lists:seq(2, N-1), is_prime(K)]),
    io:format("~w~n", [Count]).
bench_n(D) -> case os:getenv("BENCH_N") of false -> D; V -> list_to_integer(V) end.
is_prime(N) when N < 2 -> false;
is_prime(N) -> check(N, 2, trunc(math:sqrt(N))).
check(_N, D, Limit) when D > Limit -> true;
check(N, D, Limit) -> case N rem D of 0 -> false; _ -> check(N, D+1, Limit) end.
